### PR TITLE
[HDK docs] Update delete_entry & DeleteInput

### PR DESCRIPTION
**Pull request by jost-s**
### Summary
Update documentation after implementation of `DeleteInput` in #1213.

- updated docs for `delete_entry` to include `DeleteInput`
- unified descriptions for CRUD functions
- added some links to referenced functions

When this is merged, #1071 can be closed.

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [x] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/hdk/src/capability.rs [hunk 1]
```diff
@@ -175,15 +177,15 @@ pub fn generate_cap_secret() -> ExternResult<CapSecret> {
...
 ///
 /// Capability grant updates work exactly as a delete+create of the old+new grant entries.
 ///
-/// The first argument is the header hash of the old grant being deleted as per [ `delete_cap_grant` ].
-/// The second argument is the entry value of the new grant to create as per [ `create_cap_grant` ].
+/// The first argument is the header hash of the old grant being deleted as per [`delete_cap_grant`].
+/// The second argument is the entry value of the new grant to create as per [`create_cap_grant`].
 pub fn update_cap_grant(
     old_grant_header_hash: HeaderHash,
     new_grant_value: CapGrantEntry,
```
Comment (maackle) [inline crates/hdk/src/capability.rs:191]: I'm just realizing that this function probably also needs the ability to specify `ChainTopOrdering`...

File: crates/hdk/src/capability.rs [hunk 2]
```diff
@@ -94,32 +94,32 @@ pub fn create_cap_claim(cap_claim_entry: CapClaimEntry) -> ExternResult<HeaderHa
...
+/// as a [`CapClaim`] to his chain. Bob then generates a new secret and commits it in a [`CapGrant`]
 /// for `foo`, most likely explicitly `Assigned` to Alice, and sends his secret and Alice's secret
 /// to Alice's `accept_foo_grant` extern. Alice checks her grant, which matches Bob's public key
 /// and the secret Bob received from her, then she commits a new CapClaim including the secret that
 /// Bob generated. Now Alice can call `foo` on Bob's machine any time he is online, and because all
-/// the secrets are [ `CapAccess::Assigned` ] Bob can track and update exactly who has access to his externs.
+/// the secrets are [`CapAccess::Assigned`] Bob can track and update exactly who has access to his externs.
 pub fn create_cap_grant(cap_grant_entry: CapGrantEntry) -> ExternResult<HeaderHash> {
     create(CreateInput::new(
         EntryDefId::CapGrant,
```
Comment (maackle) [inline crates/hdk/src/capability.rs:125]: This fn probably also needs ability to set `ChainTopOrdering`

File: crates/hdk/src/entry.rs [hunk 3]
```diff
@@ -70,13 +72,19 @@ where
...
+/// This function accepts the [`HeaderHash`] of the element to delete and optionally an argument to
+/// specify the [`ChainTopOrdering`]. Refer to [`DeleteInput`] for details.
 ///
 /// ```ignore
 /// delete_entry(entry_hash(foo_entry)?)?;
 /// ```
+///
+/// with a specific [`ChainTopOrdering`]:
+/// ```ignore
+/// delete_entry(DeleteInput::new(entry_hash(foo_entry)?, ChainTopOrdering::Relaxed)?;
```
Comment (maackle) [inline crates/hdk/src/entry.rs:86]: Despite the name, `delete_entry` only takes HeaderHashes, not EntryHashes. Other than that, this looks good and is correct.

**Review by maackle — CHANGES_REQUESTED** [review]
Looks good, but just change the example in the docs to use header hashes. And my comments about exposing `ChainTopOrdering` to more functions is just something I noticed, not something to address here.

File: crates/hdk/src/entry.rs (hunk 3 above)
Comment (jost-s) [reply crates/hdk/src/entry.rs]: I think it's clear now.

**Review by maackle — APPROVED** [review]

**Pull request merged**